clients/src/main/java/org/apache/kafka/clients/producer/internals/RecordAccumulator.java

```diff
@@ -586,6 +642,21 @@ public boolean hasUndrained() {
         return batches;
     }
 
+    /**
+     * Bucket the batch into a bucket that's one second wide only for lookup in the
+     * inflightBatches map. This is an optimization to reduce the number of keys in the
```

Comment:
Do we have any microbenchmarks that show this (potential) optimization is justified?